examples/ota-provider-app/esp32/main/BdxOtaSender.cpp

```diff
@@ -89,6 +89,11 @@ void BdxOtaSender::HandleTransferSessionOutput(TransferSession::OutputEvent & ev
         {
             ChipLogError(BDX, "SendMessage failed: %s", chip::ErrorStr(err));
         }
+        if (event.msgTypeData.HasMessageType(chip::Protocols::SecureChannel::MsgType::StatusReport))
```

Comment:
This does not look right, for two reasons:

1. The exchange has only been closed if err == CHIP_NO_ERROR.  Otherwise we need to close it, no?
2. We should be conditioning this on the actual state that controls whether the exchange is closed: whether `sendFlags.Has(kExpectResponse)`, not on something else that happens to match up with it right now.

Reply:
Changed as per suggestion, PTAL!